UPI DL docs update - refunds (#35)
SetuHQ/docs: content/payments/upi-deeplinks/refunds.mdx

`Created`: This is when the merchant initiates a refund request via APIs or through The Bridge (Setu dashboard). The refund or its processing has not taken place at this point.

`Initiated`: This is when the refund request has been intimated to the banking partner and begins processing the refund. This is the terminal status of a refund, as NPCI does not share information about when a refund hits the customer’s bank account.

> `Pending`: This status is implied in case the refunds have not been processed due to a low balance or the merchant not transacting for a while. There are 2 reasons to elaborate on this status:

Reviewer: For a transaction that happened yesterday with a refund raised shortly after, we would be running our clearing process today. In such a case, the refund will move to an `Pending` state briefly (between the time of clearing initiation and completion), before moving to an `Initiated` state. But the current description in the PR still remains accurate for the notification section. Do you think we should clarify this difference elsewhere? (Wherein we highlight that seeing a `Pending` state in the Refund Report for example is a call for action only if the `Pending Reason` column has one of the aforementioned reasons filled in too).

Reviewer: The more I think about it, the reasons listed below are not shared in the notification payload itself. The `reason` key that we send in the notification is the _reason why/how the refund was requested in the first place_, which is particularly useful for Setu-Initiated Refunds. We can perhaps indicate that the reasons mentioned below, at least at the time of writing, are documented here for the merchant's understanding and not a part of the notification payload.

> `Pending`: This status is implied in case the refunds have not been processed due to a low balance or the merchant not transacting for a while. There are 2 reasons to elaborate on this status:

Reviewer: ```suggestion
`Pending`: This status is implied in case the refunds are awaiting to be reconciled or have completed reconciliation, yet not processed due to one of the following reasons:
```

>  - Insufficient funds: This status reason would appear if a merchant creates refunds and there is a low balance to refund all or any of the requests made. The merchant should have sufficient or more balance to avoid refunds getting delayed due to insufficient funds.

Reviewer: ```suggestion
 - Insufficient funds: Refunds requested by a merchant are adjusted from the settlement amount to the merchant's account. But if the amount for a particular refund request cannot be adjusted due to insufficient settlement amount, the refund remains in a `Pending` state.
```

>    **Example**: If the total settlement amount for the day for Merchant X is ₹850 and the merchant has created 10 refunds totalling ₹1000 in the same duration, only the first 8 will get refunded and move to an initiated state. The callback notification sent to the merchant will contain the first 8 refunds in the `initiated` block and the remaining in the `pending` block.

Reviewer: ```suggestion
   **Example**: If the total settlement amount for the day for Merchant X is ₹850 and the merchant has created 10 refunds of ₹100 each totalling ₹1000 in the same duration, only the first 8 will get refunded and move to an initiated state. The callback notification sent to the merchant will contain the first 8 refunds in the `initiated` block and the remaining in the `pending` block.
```

>    **Example**: Merchant X created 10 refunds for Rs.1000, and only 8 got refunded since the clearing balance was insufficient for the refunds to get processed, consisting of 10, 100 rupee refund requests each, and the total settlement amount for the day is 850 rupees. These would be refunded once the merchant has sufficient balance. The first 8 refunds that were registered will be processed in the current settlement cycle. A callback notification will be sent to the merchant with the payload containing the first 8 refunds in the `initiated` block and the remaining in the `pending` block.

Reviewer: ```suggestion
   **Example**: If the total settlement amount for the day for Merchant X is ₹850 and the merchant has created 10 refunds totalling ₹1000 in the same duration, only the first 8 will get refunded and move to an initiated state. The callback notification sent to the merchant will contain the first 8 refunds in the `initiated` block and the remaining in the `pending` block.
```

>  - No Active Transactions: This status implies that when a merchant has not had transactions for a while and raises a refund, active transactions are required for this to move to initiated status. 

Reviewer: ```suggestion
 - No Active Transactions: A requested refund will remain in a `Pending` state if there are no active transactions (and thus settlements to the merchant's account) from which the refund amount can be deducted and sent back to the customer.
```

   **Example**: Merchant Y has not transacted for 1 or more days and has created Refund requests. Since there are no active transactions, the refunds would remain in Pending status with the reason No Active Transactions

> 

Reviewer: ```suggestion

`Rejected`: This status is used only in the exceptional scenario where a merchant wants a previously requested refund not to be processed and requests the Setu team to reject the request. This request is currently to be raised offline.
```
